presto-main/src/main/java/io/prestosql/dispatcher/DispatchQuery.java

```diff
@@ -22,7 +22,7 @@
 {
     void recordHeartbeat();
 
-    ListenableFuture<?> getDispatchedFuture();
+    ListenableFuture<Boolean> getDispatchedFuture();
```

Comment:
Use enum instead

Reply:
i think boolean is ok, but don't feel strongly.

Reply:
Add a javadoc explaining what the future's  payload means

Reply:
I changed to enum